content/news/2022-11-05-bevy-0.9/index.md

```diff
@@ -0,0 +1,1045 @@
+
+## Scene Creation Tooling
+
+- [Create a scene from a dynamic scene][6229]
+- [dynamic scene builder][6227]
+- [can clone a scene][5855]
+
+## Improved Entity / Component APIs
+
+<div class="release-feature-authors">authors: @DJMcNab, @cart</div>
+
+Spawning entities with components and adding / removing them from entities just got even easier!
+
+First some quick fundamentals: Bevy ECS uses [`Components`][`Component`], to add data and logic to entities. To make entity composition easier, Bevy ECS also has [`Bundles`][`Bundle`], which define groups of components to be added together.
+
+Just like in previous versions of Bevy, Bundles can be tuples of components:
+
+```rust
+(Player { name: "Sova" }, Health::new(10), Team::A)
+```
+
+The [`Bundle`] trait can also be derived:
+
+```rust
+#[derive(Bundle)]
+struct PlayerBundle {
+  player: Player,
+  health: Health,
+  team: Team,
+}
+```
+
+In **Bevy 0.9**, [`Component`] types now _also_ automatically implement the [`Bundle`] trait, which allows us to consolidate all entity component operations under new `spawn`, `insert`, and `remove` apis. Previously, we had separate variants for [`Bundle`] (ex: `insert_bundle(SomeBundle)`) and [`Component`] (ex: `.insert(SomeComponent)`).
+
+The [`Bundle`] trait is now also implemented for tuples of [`Bundles`][`Bundle`] instead of just tuples of [`Components`][`Component`]. The value of this will be made clear in a moment.
+
+First: spawn now takes a bundle:
+
+```rust
+// Old (variant 1)
+commands.spawn().insert_bundle(SpriteBundle::default());
+
+// Old (variant 2)
+commands.spawn_bundle(SpriteBundle::default());
+
+// New
+commands.spawn(SpriteBundle::default());
+```
+
+Already we've saved some characters, but we're just getting started! Because [`Component`] implements [`Bundle`], we can now also pass in single components into `spawn`:
+
+```rust
+// Old
+commands.spawn().insert(Player { name: "Sova" });
+
+// New
+commands.spawn(Player { name: "Sova" });
+```
+
+Things get even more interesting when we introduce [`Bundle`] tuples into the mix, which allow us to combine many operations (covering both components and bundles) into a single `spawn` call:
+
+```rust
+// Old
+commands
+  .spawn_bundle(PlayerBundle::default())
+  .insert_bundle(TransformBundle::default())
+  .insert(ActivePlayer);
+
+// New
+commands.spawn((
+  PlayerBundle::default(),
+  TransformBundle::default(),
+  ActivePlayer,
+));
+```
+
+This is _much_ easier to type and read. And on top of that, from the perspective of Bevy ECS this is a single "bundle spawn" instead of multiple operations, which cuts down on ["archetype moves"](/news/bevy-0-5/#component-storage-the-problem). This makes this single spawn operation much more efficient!
+
+These principles apply to the insert apis as well:
+
+```rust
+// Old
+commands
+  .insert_bundle::<PlayerBundle>()
+  .insert::<ActivePlayer>();
+
+// New
+commands.insert::<(PlayerBundle, ActivePlayer)>();
```

Comment:
```suggestion
commands
  .insert_bundle(PlayerBundle::default())
  .insert(ActivePlayer);

// New
commands.insert((PlayerBundle::default(), ActivePlayer));
```

Reply:
Wow I really messed that one up didn't I :smile: